lib/transports/__es__/_base.js

```diff
@@ -36,15 +36,56 @@ class Base {
     }
```

Comment:
all these changes are unnecessary. you're introducing complexity for an untested product.